app/controllers/users_controller.rb

```diff
@@ -0,0 +1,48 @@
+class UsersController < ApplicationController
+  get '/signup' do
+    if session[:id].nil?
+      erb :'users/signup'
+    else
+      redirect '/tweets'
+    end
+  end
+
+  post '/signup' do
+    if (params[:username].empty? || params[:password].empty? || params[:email].empty?)
+      redirect '/signup'
+    else
+      user = User.create(params)
```

Comment:
if would wrap this in a build step to verify this saves to the database and offer different redirect depending on that 
```ruby 
user = User.new(params)
if user.save 
  session[:user_id] = user.id 
  redirect '/tweets'
else 
  redirect '/signup'
end 
```